core/src/main/java/io/onetable/delta/DeltaSchemaExtractor.java

```diff
@@ -218,7 +216,6 @@ private OneSchema toOneSchema(
         break;
       case "decimal":
         DecimalType decimalType = (DecimalType) dataType;
-        metadata = new HashMap<>(2, 1.0f);
```

Comment:
This is creating a map of size 2 allocated with scale factor 1.0f to make sure we don't go through a resize operation. How much are we saving with EnumMap with 5 keys? The number of keys may grow but I think it is safe to assume that we'll keep it below 10 for the near future.